### PR TITLE
Tests validateDifficulty ( ValidatePruningPointViolationAndProofOfWorkAndDifficulty)

### PR DESCRIPTION
**Pull request by talelbaz**
part of #1314 
(and also part of #1008)
Adds tests for validateDifficulty on ValidatePruningPointViolationAndProofOfWorkAndDifficulty.go

### CODE REVIEW

File: domain/consensus/processes/blockvalidator/pruning_violation_proof_of_work_and_difficulty_test.go [hunk 1]
```diff
@@ -241,3 +242,59 @@ func TestCheckPruningPointViolation(t *testing.T) {
...
+		mocDifficulty := &mocDifficultyManager{}
+		factory.SetTestDifficultyManager(func(model.DBReader, model.GHOSTDAGManager, model.GHOSTDAGDataStore,
+			model.BlockHeaderStore, model.DAGTopologyManager, model.DAGTraversalManager, *big.Int, int, bool, time.Duration,
+			*externalapi.DomainHash) model.DifficultyManager {
+			return mocDifficulty
+		})
+		genesisDifficulty := params.GenesisBlock.Header.Bits()
+		mocDifficulty.testDifficulty = genesisDifficulty
+		mocDifficulty.testGenesisBits = genesisDifficulty
+		tc, teardown, err := factory.NewTestConsensus(params, false, "TestCheckPruningPointViolation")
```
Comment (someone235) [inline domain/consensus/processes/blockvalidator/pruning_violation_proof_of_work_and_difficulty_test.go:259]: Please fix TestCheckPruningPointViolation->TestValidateDifficulty

File: domain/consensus/processes/blockvalidator/pruning_violation_proof_of_work_and_difficulty_test.go [hunk 2]
```diff
@@ -241,3 +242,59 @@ func TestCheckPruningPointViolation(t *testing.T) {
 		}
 	})
 }
+
+func TestValidateDifficulty(t *testing.T) {
```
Comment (someone235) [inline domain/consensus/processes/blockvalidator/pruning_violation_proof_of_work_and_difficulty_test.go:246]: Can you please add a comment that explains what the test does?

**Review by someone235 — CHANGES_REQUESTED** [review]

File: domain/consensus/processes/blockvalidator/pruning_violation_proof_of_work_and_difficulty_test.go [hunk 3]
```diff
@@ -241,3 +242,64 @@ func TestCheckPruningPointViolation(t *testing.T) {
...
+}
+
+type mocDifficultyManager struct {
+	testDifficulty  uint32
+	testGenesisBits uint32
+}
+
+func (dm *mocDifficultyManager) genesisBits() (uint32, error) {
+	return dm.testGenesisBits, nil
+}
```
Comment (elichai) [inline domain/consensus/processes/blockvalidator/pruning_violation_proof_of_work_and_difficulty_test.go:300]: Why is this needed?

**Review by elichai — DISMISSED** [review]

File: domain/consensus/processes/blockvalidator/pruning_violation_proof_of_work_and_difficulty_test.go [hunk 4]
```diff
@@ -243,6 +243,11 @@ func TestCheckPruningPointViolation(t *testing.T) {
 	})
 }
 
+// TestValidateDifficulty verified that in case of a block with an unexpected difficulty,
```
Comment (someone235) [inline domain/consensus/processes/blockvalidator/pruning_violation_proof_of_work_and_difficulty_test.go:246]: verified->verifies

File: domain/consensus/processes/blockvalidator/pruning_violation_proof_of_work_and_difficulty_test.go [hunk 5]
```diff
@@ -243,6 +243,11 @@ func TestCheckPruningPointViolation(t *testing.T) {
...
+// an appropriate error message (ErrUnexpectedDifficulty) will be returned on the
+// function ValidatePruningPointViolationAndProofOfWorkAndDifficulty. The require difficulty is
```
Comment (someone235) [inline domain/consensus/processes/blockvalidator/pruning_violation_proof_of_work_and_difficulty_test.go:248]: require->required

**Review by someone235 — CHANGES_REQUESTED** [review]

**Review by someone235 — APPROVED** [review]

**Pull request merged**